src/SimpleAjax/SimpleAjax.php

```diff
@@ -82,31 +81,41 @@ public function __construct()
         \Controller::setStaticUrls();
     }
 
-
     /**
      * Handle the request
      */
     public function handle()
     {
         global $container;
+
         /** @var EventDispatcher $dispatcher */
         $dispatcher = $container['event-dispatcher'];
 
         // Trigger event
+        // @since 1.1
         $event = new SimpleAjaxEvent($this->isIncludeFrontendExclusive());
         $dispatcher->dispatch(SimpleAjaxEvent::NAME, $event);
 
+        // If the event listener does not terminate the process by itself, check for a response instance to send
+        // @since 1.2
+        $response = $event->getResponse();
+        if (null !== $response) {
```

Comment:
Wouldn't all of this be more streamlined if you would do:
```php
// If the event listener does not terminate the process by itself, check for a response instance to send
// @since 1.2
if (null === ($response = $event->getResponse()) {
    // Run hooks
    $this->runHooks();
    // Hooks should have been exiting now - if we end up, prepare an error response.
    $response = new Response(
        'Simple Ajax: Invalid AJAX call.',
        Response::HTTP_PRECONDITION_FAILED
       );
}
$response->send();
```
IMO this is more readable than trying to send the response twice.
Especially as the HOOK approach is legacy code.